src/_gettsim/functions_loader.py

```diff
@@ -571,6 +606,129 @@ def aggregation_func(source_col, group_id):
+    link_functions = {
+        link_col: _create_one_link_func(
+            link_spec,
+            user_and_internal_functions,
+            data_cols,
+        )
+        for link_col, link_spec in link_dict.items()
+        if link_spec["source_col"] in user_and_internal_functions
+    }
+
+    return link_functions
+
+
+def _create_one_link_func(
+    link_spec: dict[str, str],
+    user_and_internal_functions: dict[str, Callable],
+    data_cols: list[str],
+) -> Callable:
+    """Create a function that links parent and child variables."""
+
+    _fail_if_source_col_not_in_functions(
+        link_spec, user_and_internal_functions, data_cols
+    )
+
+    annotations = {}
+
+    if (
+        link_spec["source_col"] in user_and_internal_functions
+        and "return"
+        in user_and_internal_functions[link_spec["source_col"]].__annotations__
+    ):
+        annotations["source_col"] = user_and_internal_functions[
+            link_spec["source_col"]
+        ].__annotations__["return"]
+        annotations["return"] = (
+            int if annotations["source_col"] in (int, bool) else float
+        )
+    else:
+        pass
+
+    # Single target
+    if isinstance(link_spec["id_col"], str):
+        # Create id annotations
+        annotations["id_col"] = int
+        # Create function
+        mapper_args = {
+            "id_col": link_spec["id_col"],
+            "p_id": "p_id",
+            "source_col": link_spec["source_col"],
+        }
+
+        @rename_arguments(mapper=mapper_args, annotations=annotations)
+        def link_func(
+            source_col: numpy.ndarray,
+            id_col: numpy.ndarray,
+            p_id: numpy.ndarray,
+        ) -> numpy.ndarray:
+            return sum_by_parent(source_col, id_col, p_id)
+
+    # Multiple targets
+    elif len(link_spec["id_col"]) == 2:
+        # Create id annotations
+        annotations.update({"id_col_1": int, "id_col_2": int})
+        # Create function
+        mapper_args = {
+            "id_col_1": link_spec["id_col"][0],
+            "id_col_2": link_spec["id_col"][1],
+            "p_id": "p_id",
+            "source_col": link_spec["source_col"],
+        }
+
+        @rename_arguments(mapper=mapper_args, annotations=annotations)
+        def link_func(
+            source_col: numpy.ndarray,
+            id_col_1: numpy.ndarray,
+            id_col_2: numpy.ndarray,
+            p_id: numpy.ndarray,
+        ) -> numpy.ndarray:
+            return sum_by_parent_multiple_targets(source_col, id_col_1, id_col_2, p_id)
+
+    else:
+        raise NotImplementedError(
+            """Parent-child links based on more
+                                  than two ID columns currently not supported."""
+        )
+
+    return link_func
+
+
+def sum_by_parent(
+    source_col: numpy.ndarray,
+    id_col: numpy.ndarray,
+    p_id: numpy.ndarray,
+) -> numpy.ndarray:
+    return sum_values_by_index(source_col, id_col, p_id)
+
+
+def sum_by_parent_multiple_targets(
```

Comment:
See above, though here the interface changes, of course. Might still be clearer to do it just in the calling code directly, rather than adding a function.